src/openldap/mod.rs

```diff
@@ -0,0 +1,519 @@
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct AccesslogSettings {
+    log_operations: AccesslogLogOperations,
+    log_success: bool,
+    log_purge: (String, String),
+    log_old: String,
+    log_old_attribute: String,
+}
+impl AccesslogSettings {
+    /// Specify which types of operations to log.
+    /// Default: [`AccesslogLogOperations::Writes`]
+    pub fn with_log_operations(mut self, log_operations: AccesslogLogOperations) -> Self {
+        self.log_operations = log_operations;
+        self
+    }
+    /// Whether successful operations should be logged
+    /// Default: `true`
+    pub fn with_log_success(mut self, log_success: bool) -> Self {
+        self.log_success = log_success;
+        self
+    }
+    /// When and how often old access log entries should be purged. Format "dd+hh:mm".
+    /// Default: `("07+00:00", "01+00:00")`.
+    pub fn with_log_purge(mut self, (when_log_purge, how_often_log_purge): (&str, &str)) -> Self {
+        self.log_purge = (when_log_purge.to_owned(), how_often_log_purge.to_owned());
+        self
+    }
+    /// An LDAP filter that determines which entries should be logged.
+    /// Default: `"(objectClass=*)"`
+    pub fn with_log_old(mut self, log_old: &str) -> Self {
+        self.log_old = log_old.to_owned();
+        self
+    }
+    /// Specifies an attribute that should be logged.
+    /// Default: `"objectClass"`.
+    pub fn with_log_old_attribute(mut self, log_old_attribute: &str) -> Self {
+        self.log_old_attribute = log_old_attribute.to_owned();
+        self
+    }
+}
+
+impl Default for AccesslogSettings {
+    fn default() -> Self {
+        Self {
+            log_operations: AccesslogLogOperations::Writes,
+            log_success: true,
+            log_purge: ("07+00:00".to_owned(), "01+00:00".to_owned()),
+            log_old: "(objectClass=*)".to_owned(),
+            log_old_attribute: "objectClass".to_owned(),
+        }
+    }
+}
+
+impl Default for OpenLDAP {
+    /// Starts an instance with horrible passwords values.
+    /// Obviously not to be emulated in production.
+    ///
+    /// Defaults to:
+    /// - Admin: (username: `"admin"`, password: `"adminpassword"`, dn: `"cn=admin,dc=example,dc=org"`)
+    /// - Users: `[]`
+    /// - Accesslog admin: `None`
+    /// - Anonymous bindings: `true`
+    fn default() -> Self {
+        Self {
+            users: vec![],
+            env_vars: HashMap::new(),
+        }
+    }
+}
+
+impl Image for OpenLDAP {
+    fn name(&self) -> &str {
+        NAME
+    }
+
+    fn tag(&self) -> &str {
+        TAG
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        // maybe OpenLDAP will have a healthcheck someday
+        // https://github.com/osixia/docker-openldap/issues/637
+        vec![
+            WaitFor::message_on_stderr("** Starting slapd **"),
+            WaitFor::seconds(2), // ideally, one should wait for a port instead of waiting a fixed time
```

Comment:
Probably we can utilize HTTP wait strategy? 🤔

Reply:
I see, thank you

Yes, let's use delay for now and later we can add the strategy